evals/elsuite/tools/eval.py

```diff
@@ -67,15 +68,34 @@ def eval_sample(self, sample: Sample, rng):
             logging.info(f"Prompt: {prompt}")
             logging.info(f"Error: {str(e)}")
             sampled = "ERROR: " + str(e)
-        match = sampled == assistant_msg["content"]
-        evals.record.record_match(match, expected=assistant_msg["content"], sampled=sampled)
-        return match
+
+        match = self.match_tool_call(assistant_msg, sampled)
+        evals.record.record_metrics(score=match)
+
+    def match_tool_call(self, gt_message: Sample, sampled: List[Union[str, Dict[str, Any]]]) -> int:
+        sampled_tool_call = isinstance(sampled, dict) and sampled.get("type") == "function"
+        expected_tool_call = gt_message.get("tool_calls") is not None
+
+        if sampled_tool_call != expected_tool_call:
+            return 0
+
+        if not sampled_tool_call:
+            # simply not using a tool call is considered a valid response
+            return 1
+
+        sampled_func = sampled["function"]
+        gt_func = gt_message["tool_calls"][0]["function"]
+
+        return (
+            sampled_func["name"] == gt_func["name"]
+            and sampled_func["arguments"] == gt_func["arguments"]
+        )
```

Comment:
Alternative score:

```suggestion
        score = (
            1  # since it did a tool call
            + (sampled_func["name"] == gt_func["name"])
            + (sampled_func["arguments"] == gt_func["arguments"])
        )

        return score / 3
```